docs/proposals/05-etcd-operator-tasks.md

```diff
@@ -0,0 +1,369 @@
+  // Type of on-demand snapshot.
+  SnapshotType snapshotType `json:"snapshotType"`
+}
+```
+
+##### Pre-Conditions
+
+* Etcd cluster should have a quorum.
+* There should not already be a `on-demand snapshot` task running with the same `SnapshotType` for the same etcd cluster.
+
+#### Trigger on-demand maintenance of etcd cluster
+
+Operator can trigger on-demand maintenance of etcd cluster which includes operations like [etcd compaction](https://etcd.io/docs/v3.5/op-guide/maintenance/#history-compaction-v3-api-key-value-database), [etcd defragmentation](https://etcd.io/docs/v3.2/op-guide/maintenance/#defragmentation) etc.
+
+##### Possible Scenarios
+
+* If an etcd cluster is heavily loaded which is causing performance degradation of an etcd cluster and the operator does not want to wait for the scheduled maintenance window then an on-demand maintenance task can be triggered which will invoke etcd-compaction, etcd-defragmentation etc. on the target etcd cluster. This will make etcd cluster lean and clean, thus improving cluster performance.
+
+##### Task Config
+
+```go
+type OnDemandMaintenanceTaskConfig struct {
+  // MaintenanceType defines the maintenance operations need to be performed on etcd cluster.
+  MaintenanceType maintenanceOps
+}
+
+type maintenanceOps struct {
+  EtcdCompaction *bool `json:"etcd-compaction,omitempty"`
+  EtcdDefragmentation *bool `json:"defragmentation,omitempty"`
+}
+```
+
+##### Pre-Conditions
+
+* Etcd cluster should have a quorum.
+* There should not already be a duplicate task running with same `maintenanceType`.
+
+#### Copy the backups from one object store to another object store
+
+##### Task Config
+
+```go
+// EtcdCopyBackupsTaskConfig defines the parameters for the copy backups task.
+type EtcdCopyBackupsTaskConfig struct {
+  // SourceStore defines the specification of the source object store provider.
+  SourceStore StoreSpec `json:"sourceStore"`
+
+  // TargetStore defines the specification of the target object store provider for storing backups.
+  TargetStore StoreSpec `json:"targetStore"`
+
+  // MaxBackupAge is the maximum age in days that a backup must have in order to be copied.
+  // By default all backups will be copied.
+  // +optional
+  MaxBackupAge *uint32 `json:"maxBackupAge,omitempty"`
+
+  // MaxBackups is the maximum number of backups that will be copied starting with the most recent ones.
+  // +optional
+  MaxBackups *uint32 `json:"maxBackups,omitempty"`
+}
+```
+
+##### Pre-Conditions
+
+* There should not already be a duplicate task running.
+
+> Note: `copy-backup-task` runs as a seprate job, hence it doesn't depends on health of etcd cluster members or any other conditions.
+
+> Note: CopyBackupTask has already been implemented and it's currently being used in [Control Plane Migration](https://github.com/gardener/gardener/blob/master/docs/proposals/07-shoot-control-plane-migration.md) but CopyBackupTask can be harmonize with `EtcdOperatorTask` custom resource.
```

Comment:
Does this mean we need to deprecate the `EtcdCopyBackupsTask` resource in favour of the harmonized `EtcdOperatorTask` resource? If so, how do we plan to do this? If not, would druid need to support both ways of invoking a copy-backups task?

Reply:
I think we need to deprecate the old way of invoking the `EtcdCopyBackupsTask` but at some of time we need to support both ways as CPM code flow also need to be adjust accordingly.